library/hls/src/main/java/com/google/android/exoplayer2/source/hls/HlsSampleStreamWrapper.java

```diff
@@ -849,6 +937,8 @@ public TrackOutput track(int id, int type) {
     sampleQueueTrackIds[trackCount] = id;
     sampleQueues = Arrays.copyOf(sampleQueues, trackCount + 1);
     sampleQueues[trackCount] = trackOutput;
+    HlsMediaChunk mediaChunk = findChunkMatching(chunkUid);
+    mediaChunk.setFirstSampleIndex(trackCount, 0);
     sampleQueueReaders = Arrays.copyOf(sampleQueueReaders, trackCount + 1);
```

Comment:
When each elementary stream is discovered and the SampleQueue is created, this is needed for the first chunks loaded right after the initial Extractor is created.